lib/dolly/mango_query.rb

```diff
@@ -0,0 +1,37 @@
+require 'dolly/mango/selector'
+module Dolly
+  class MangoQuery
+    include Dolly::Mango::Selector
+
+    FIELDS_KEY   = 'fields'.freeze
+    SORT_KEY     = 'sort'.freeze
+    LIMIT_KEY    = 'limit'.freeze
+
+    attr_reader :proxy_class, :query
+
+    def initialize proxy_class
+      @proxy_class = proxy_class
+      @query = Hash.new{ |h,k| h[k] = Hash.new(&h.default_proc) }
```

Comment:
can you add that first level `[SELECTOR]` here? and remove it from all the `build_` methods? 
something like..
`Hash.new{ |h,k| h[SELECTOR][k] = Hash.new(&h.default_proc) }`
and then remove the selector from these calls. 
`@query[SELECTOR][name][LTE_OPERATOR] = value` => `@query[name][LTE_OPERATOR] = value`

not sure if its a good idea or if it will work.. just pointing it out.  ignore if it will stay